v2/pkg/engine/resolve/resolvable.go

```diff
@@ -703,7 +703,7 @@ func (r *Resolvable) addRejectFieldError(reason string, ds DataSourceInfo, field
 	} else {
 		errorMessage = fmt.Sprintf("Unauthorized to load field '%s', Reason: %s.", fieldPath, reason)
 	}
-	r.ctx.appendSubgraphError(goerrors.Join(errors.New(errorMessage),
+	r.ctx.appendError(goerrors.Join(errors.New(errorMessage),
```

Comment:
`appendError` calls `errors.Join` inside, which takes a variadic second parameter, perhaps we could just pass this through to avoid nesting `errors.Join` calls and this odd `goerrors` alias.